### ISSUE TITLE
Readme states this module is not useful v7 onwards...

### ISSUE THREAD

**Issue by jsumners** [mcollina/fast-json-parse#6]
... but it does not state if one should just rely on `JSON.parse` or if another module is more appropriate. It would help to be more explicit in the readme.

**Comment by mcollina** [comment]
This module is totally superseded by turbofan. It should not be used.

**Comment by jcao219** [comment]
Is it possible, theoretically, to optimize `JSON.parse` if you know the schema ahead of time? Just a curiosity of mine.

**Comment by mcollina** [comment]
You can’t because properties in objects can appear in every order.

**Comment by jcao219** [comment]
True, but objects in JSON strings always appear in the order that they appear.

If they follow the order of, say, the schema object (in JS the order is chronological except for 1 exception due to ES2015, according to [this](https://www.stefanjudis.com/today-i-learned/property-order-is-predictable-in-javascript-objects-since-es2015/)).

I believe it could be a theoretical pathway for optimization. You would have to be strict on the JSON (basically, JSONSchema +ordering) but then it can be optimized.

**Comment by mcollina** [comment]
I’ve tried several times. It’s likely impossible to improve on JSON.parse() in any way, mainly because it has been incredibly optimized in V8.

**Comment by jcao219** [comment]
I wouldn't say so...

There are some optimizations that you can do, if you know the shape of the JSON with near certainty.

For instance, for JSON with a lot of objects with only singleton properties, e.g.
`{"error":{"type":{"StackError":{"stack":"....","stackDetails":{ "detail":{}}}}}`

if you know that `error` is always a singleton property, you can optimize the parsing of
`{ error: [ErrObj]}`

into just:

`json => ({ error: JSON.parse(json.slice(4 + "error".length, -1))})`

I'll have to ask someone on the V8 team if this makes sense.

Obviously this is highly limited to this:

* You know the JSON Schema
* You know the order of the properties
* No extra whitespace

But I'm sure it's possible, as a thought experiment :-)  not sure if it would have practical benefits.

JSON parsing is already way absurdly faster than it had any right to be, thanks to v8

-- edit:

decided to play with the idea:
https://github.com/jcao219/json-parse-tricks/blob/master/src/index.ts

I think it's very possible to optimize, but its usefulness is highly limited to these:
* JSON Schema known
* JSON schema only has fixed length property values
* (even better) knowing the exact order of the JSON (e.g. alphabetical, arbitrary, etc)

**Comment by phr3nzy** [comment]
Check this https://github.com/mafintosh/turbo-json-parse
